File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public List<Integer> solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList<>();
        Queue<Integer> queue = new ArrayDeque<>();
        
        // queue에 남은 일수 저장
        for (int i = 0; i < progresses.length; i++) {
            int days = (100 - progresses[i] + speeds[i] - 1) / speeds[i];
            queue.offer(days);
        }
        
        // while 문 안에 while 문이 있지만 실제로는 한번씩만 접근해서 O(n)
        while (!queue.isEmpty()) {
            int current = queue.poll();
            int count = 1;
            
            while (!queue.isEmpty() && queue.peek() <= current) {
                queue.poll();
                count++;
            }
            answer.add(count);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    length = len(progresses)
    
    deploy = [0] * length
    
    for i in range(length):
        if (100 - progresses[i]) % speeds[i] == 0:
            deploy[i] = int((100 - progresses[i]) / speeds[i])
        else:
            deploy[i] = int((100 - progresses[i]) // speeds[i]) + 1
    
    day = deploy[0]
    count = 1
    for i in range(1, length):
        if deploy[i] <= day:
            count += 1
        else:
            day = deploy[i]
            answer.append(count)
            count = 1
        if i == length-1:
            answer.append(count)
    
    return answer